net/lp_rpc.proto

```diff
@@ -20,12 +20,51 @@ message TranscoderRequest {
   bytes sig   = 2;
 }
 
+/*
+  OSInfo needed to negotiate storages that will be used.
+  It carries info needed to write to the storage.
+*/
+message OSInfo {
+
+  enum StorageType {
+    DIRECT = 0;
+    S3     = 1;
+    IPFS   = 2;
+  }
+
+  // Storage type: direct, s3, ipfs.
+  StorageType storageType = 1;
+
+  S3OSInfo s3info = 16;
+}
+
+message S3OSInfo {
+
+  // Host to use to connect to S3
+  string host = 1;
+
+  // Path (prefix) to use when uploading the object.
+  string path = 2;
```

Comment:
why is `path` part of `S3OSInfo`? Path is part of `OSDriver.NewSession(path string)` interface, so path should be in general `OSInfo` class, not in specialized S3 class?

Reply:
Renamed the field to "key" and pushed, let me know.